pkg/datastreams/live_messages.go

```diff
@@ -0,0 +1,169 @@
+}
+
+// String returns the name of the provider.  All Config instances produced
+// by this provider will have this value in their Provider field.
+func (c *Controller) String() string {
+	return names.DataStreamsLiveMessages
+}
+
+// GetConfigErrors returns a map of errors that occurred on the last Collect
+// call, indexed by a description of the resource that generated the error.
+// The result is displayed in diagnostic tools such as `agent status`.
+func (c *Controller) GetConfigErrors() map[string]providers.ErrorMsgSet {
+	return nil
+}
+
+// ManageSubscriptionToRC subscribes to remote configuration updates if the agent is running the kafka_consumer integration.
+func (c *Controller) ManageSubscriptionToRC(subscribe func()) {
+	ticker := time.NewTicker(time.Second * 10)
+	defer ticker.Stop()
+	for range ticker.C {
+		c.closeMutex.RLock()
+		if c.closed {
+			c.closeMutex.RUnlock()
+			return
+		}
+		c.closeMutex.RUnlock()
+		if isConnectedToKafka(c.ac) {
+			subscribe()
+			return
+		}
+	}
+}
+
+func isConnectedToKafka(ac autodiscovery.Component) bool {
+	for _, config := range ac.GetAllConfigs() {
+		if config.Name == kafkaConsumerIntegrationName {
+			return true
+		}
+	}
+	return false
+}
+
+// NewController creates a new Controller instance
+func NewController(ac autodiscovery.Component) *Controller {
+	return &Controller{
+		ac:            ac,
+		configChanges: make(chan integration.ConfigChanges, 10),
+	}
+}
+
+// Stream starts sending configuration updates for the kafka_consumer integration to the output channel.
+func (c *Controller) Stream(ctx context.Context) <-chan integration.ConfigChanges {
+	go func() {
+		<-ctx.Done()
+		c.closeMutex.Lock()
+		defer c.closeMutex.Unlock()
+		c.closed = true
+		close(c.configChanges)
+	}()
+	return c.configChanges
+}
+
+// Update parses updates from remote configuration, and configures the kafka_consumer integration to fetch messages from Kafka
+func (c *Controller) Update(updates map[string]state.RawConfig, applyStateCallback func(string, state.ApplyStatus)) {
+	remoteConfigs := parseRemoteConfig(updates, applyStateCallback)
+	if len(remoteConfigs) == 0 {
+		return
+	}
+	configChange := integration.ConfigChanges{}
+	cfgs := c.ac.GetAllConfigs()
+	for _, integrationConfig := range cfgs {
+		if integrationConfig.Name != kafkaConsumerIntegrationName {
+			continue
+		}
+		configChange.Unschedule = append(configChange.Unschedule, integrationConfig)
+		updatedConfig := integrationConfig
+		updatedConfig.Instances = make([]integration.Data, 0, len(updatedConfig.Instances))
+		if updatedConfig.LogsConfig == nil {
+			updatedConfig.LogsConfig = integration.Data("[{\"type\":\"integration\",\"service\":\"kafka_consumer\",\"source\":\"kafka_consumer\"}]")
+		}
+		for _, instance := range integrationConfig.Instances {
+			updatedInstance := instance
+			p := &updatedInstance
+			err := p.SetField("live_messages_configs", remoteConfigs)
+			if err != nil {
+				log.Error("Error setting field")
```

Comment:
```suggestion
				log.Error("Live messages update: Error setting field")
```

Reply:
Also, for my understanding, if we can not set `live_messages_configs`, would that mean that the new integrations would not log, right?

Reply:
Correct. Nothing will be logged in this case.